### ISSUE TITLE
Error out when axe.configure is given bad values

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2282]
QA stumbled upon this issue by accident. The following code creates a rule with an undefined ID, which is really odd:

```js
axe.configure({ rules: ['foo'] })
```

The content of the rules array should be an object, not a string, but that's never caught. At the very least axe-core should throw when the id of a check / rule is undefined. The ID has to be a string.

**Comment by jeankaplansky** [comment]
no user-facing does required.
